### PR TITLE
First implementation of the blog for the website update

### PR DESCRIPTION
**Pull request by atxr**
This is my first contribution to the new website.
This branch implements the main functionalities of the blog for the website. 

Critical functionalities are still missing, but I ask reviews for typos, suggestions and bugs on functions that are already implemented. Testing the application on multiple web browsers can also be a good point.

I also ask @J3y0 and @Turtyo to perform simple security checks regarding the following critical points:
- File upload
- No SQL injection

Once this PR will be merged, I hope you will create a lot of issues for incoming functionalities.

### CODE REVIEW

File: .gitignore
Comment (amtoine) [reply .gitignore]: got it :ok_hand:

File: .gitignore
Comment (amtoine) [reply .gitignore]: do not see these files now in the `git status`, that's good for me if they do not need to be tracked :yum: 
and i see the `package` files :ok_hand:

File: frontend/.gitignore
Comment (amtoine) [reply frontend/.gitignore:16]: valid reason not to touch them :+1:

File: frontend/public/manifest.json
Comment (amtoine) [reply frontend/public/manifest.json:1]: :ok_hand:

File: frontend/public/robots.txt
Comment (amtoine) [reply frontend/public/robots.txt:1]: got it :wink:

File: backend/.env_example
Comment (amtoine) [reply backend/.env_example]: perfect for the database and the comments, thanks :ok_hand:

**Review by amtoine — APPROVED** [review]

**Comment by amtoine** [comment]
Thanks for the changes, that looks really good now :yum: 
i'll have a look at the source code later, no time for now :wink:

**Review by amtoine — COMMENTED** [review]
I had a look at the source code :+1: 

well... that is quite though as a first experience with `javascript` :open_mouth: 
not a lot to say on the code itself, to be honest :thinking: 

maybe some suggestions:
1. add comments inside the code base, e.g. when functions are defined, explain what they do
2. from my `python` experience, i like two empty lines around function definitions, no idea if it's accepted in `javascript`
3. is it possible to move some of the hardcoded paths and names to a config file? or maybe that would never change and that's not a big deal!
4. this is a minor thing, but what about adding "/" at the end of the collapsed directories in the main `README`? e.g. `├── pages/` instead of `├── pages`? just to make it clear that there are files in there to check out :+1: 

in the end and overall, if you find my suggestions relevant in the scope of that PR, i think the source code is great and works fine as a first version of the `iscsc.fr` website :muscle:

**Comment by atxr** [comment]
Thanks for your review @amtoine !
1. I'll definitely consider it for the next time, but I think I will let the code like this for this PR
2. I use a `js` code formatter (`prettier`) so I guess it's standard `js`
3. Do you have an example in mind?
4. Yes sure if it helps!
Should we wait for another review before merging?

**Comment by amtoine** [comment]
@atxr

1. yup, inside another PR is fine
2. fine with me if that's the `js` formatter :ok_hand: 
3. for instance, in `./frontend/src/App.js`, there are all these hardcoded relative paths
```js
import Blog from "./pages/blog/Blog";
import Navbar from "./components/Navbar";
import CreateArticle from "./pages/blog/CreateArticle";
import Article from "./pages/blog/Article";
```
i was just wondering if we could move these to come centralized config file
i'm just not a huge fan of hardcoded constants, but that might be the good way to go with this project, you tell me :yum: 
4. waiting for the change then :wink: 

> Should we wait for another review before merging?
>
i think that would be better but this PR is a bit the default start and people do not look super hyped by your code for now :eyes:
it's fine with me to merge now after you change the `README` maybe, but we agreed to do 2 reviews for each PR at the beginning... :thinking:

**Comment by atxr** [comment]
3. Actually it's very classical React code so I won't change it. It's like java or C(++) imports, you have to put the relative path depending of your project structure.
4. ✔️ 
I will try to motivate another guy to at least run the website then! Otherwise ready for the merge 🚀

**Comment by amtoine** [comment]
> 3. Actually it's very classical React code so I won't change it. It's like java or C(++) imports, you have to put the relative path depending of your project structure.
>
legit reason to me :yum: 

> 4. heavy_check_mark
>    I will try to motivate another guy to at least run the website then! Otherwise ready for the merge rocket
>
lookin' forward to it :relieved:

**Review by amtoine — APPROVED** [review]
Thanks fot the `/` in the `README` :+1:

**Comment by Turtyo** [comment]
I was able to run the website without problem, apart from warning messages on old methods used that are now obsolete on the frontend part.

I think a little `website_init.sh` could be a great think to add, it's not very long to launch the website by hand but if we can gain on that...

For the website in itself, I had no problem reading articles or writing one. The markdown didn't specify how to add a newline without a blank space but I found it was with a `\` after a bit of testing.
Overall it works well, I guess features such as editing will be added later. Another thing that would be good is to expand the window where we put the text, because it's not very large nor very high and there seems to be room left for that on the page.

Since i don't see a problem and I'm just suggesting modifications (that should be discussed in a issue anyway), I'll approve the PR

**Review by ctmbl — COMMENTED** [review]
Great PR! You've done an amazing work for the website :heart:
Sorry for the late review I was really busy during August :confused: 

Some comments still:
 - is CamelCase widely used for URL links like `/createArticle`? I don't know at all web conventions
 - commit bodies are quite useful and mandatory when the commit is a bit technical, especially because we're not React experts
 - a real thanks for your beautiful and incremental commits it made the review far easier!
 - making the README.md along the way of coding the site is really useful as well to bring precision on the modifications and putting some concrete things on abstract code concepts (once again because we don't know React)
 - maybe the only thing that is really missing here is a global note/guide on 
   - React concepts or documentation
   - interactions between MongoDB React and other techs
   - basic description of the website overall, components, file structure etc
 It seems quite mandatory to me to be able to better understand every piece of code
 
Because this PR is now closed let's not be long here, simply answer me what points could be interesting to develop and I'll write some issues!

**Comment by atxr** [comment]
> Great PR! You've done an amazing work for the website heart Sorry for the late review I was really busy during August confused
> 
Thank you and no problem!

>     * is CamelCase widely used for URL links like `/createArticle`? I don't know at all web conventions

That's a good point it should be /create-article

>     * commit bodies are quite useful and mandatory when the commit is a bit technical, especially because we're not React experts
>     * a real thanks for your beautiful and incremental commits it made the review far easier!

Thanks! I'll continue this way then!

>     * making the README.md along the way of coding the site is really useful as well as bringing precision on the modifications and putting some concrete things on abstract code concepts (once again because we don't know React) but as @amtoine said commenting the code could be great too!

Sure, I'll remember this later

>     * maybe the only thing that is really missing here is a global note/guide on
>       
>       * React concepts or documentation
>       * interactions between MongoDB React and other techs
>       * basic description of the website overall, components, file structure etc
>         It seems quite mandatory to me to be able to better understand every piece of code

Mmm about this I'm thinking about redacting a article on the website itself to explain a bit the MERN stack for those who aren't familiar with web technologies,
But I can also add it to the README for example!
 
> Because this PR is now closed let's not be long here, simply answer me what points could be interesting to develop and I'll write some issues!
Yes you can definitely create an issue for the first point then!

File: frontend/src/App.js
Comment (atxr) [reply frontend/src/App.js:11]: Actually, I'm closing the tag.
It's like doing
```html
<img src="foo.png" />
```
It's a classical way to write tags when they don't have children

File: frontend/src/index.css
Comment (atxr) [reply frontend/src/index.css]: Mmm I don't really know.
Adding the colors to a config file is a bit confusing because CSS files are supposed to be style config files.
But I tried to put some CSS variable at the beginning of the file to avoid hard-coding the colors in the style file.

In index.css:
```css
:root {
  --primary: #1aac83;
  --secondary: #f1f1f1;
  --error: #e7195a;
}
```

But maybe I missed some of them!

File: frontend/src/components/ArticleForm.js
Comment (atxr) [reply frontend/src/components/ArticleForm.js:27]: Yes it should definitely be a else

File: frontend/src/index.css
Comment (atxr) [reply frontend/src/index.css:138]: I had some duplicates in the class names...
But I will try to refactor the CSS file it's a real mess
Maybe I will use something like [tailwind](https://tailwindcss.com/)

**Comment by atxr** [comment]
I don't know if I should modify the files now are wait for the end of #2

File: README.md
Comment (ctmbl) [reply README.md:43]: Ok should I propose such a change in #2 ?

File: frontend/src/App.js
Comment (ctmbl) [reply frontend/src/App.js:11]: Ok now I understand tks!

File: frontend/src/components/ArticleForm.js
Comment (ctmbl) [reply frontend/src/components/ArticleForm.js:11]: Ok tks for the explanation!

File: frontend/src/components/ArticleForm.js
Comment (ctmbl) [reply frontend/src/components/ArticleForm.js:27]: Should I propose a change or should we let it has it is?

File: frontend/src/index.css
Comment (ctmbl) [reply frontend/src/index.css:138]: Ok fine!

**Review by ctmbl — COMMENTED** [review]
Tks for your explanations it helped a lot!

File: frontend/src/index.css
Comment (ctmbl) [reply frontend/src/index.css]: Ok fine for me!

**Comment by ctmbl** [comment]
> I don't know if I should modify the files now are wait for the end of #2

We should propose a #3 PR I think, whether we merge it before or after #2 isn't a problem!

**Comment by atxr** [comment]
You can put all your recommendations in #2 it will be easier!

**Comment by amtoine** [comment]
@ctmbl @atxr :wave: 

#2 is about authentification to the Mongo database :thinking: 
i think the new comments above are out of the scope of #2 and thus should end up in a separate issue :yum:

**Pull request merged**
